### ISSUE TITLE
[Feature Request][SF][1.28.4][List][A11y][Expected Fixed Time: ASAP] Cannot change the title heading level

### ISSUE THREAD

**Issue by wilson-xu-sap** [SAP/ui5-webcomponents#9652]
### Feature Request Description

Cannot change title heading level when use `titleText`, which violates the a11y standard.

### Proposed Solution

Add a new prop for setting and calculating the heading level.

### Organization

SF

### Priority

Medium

### Privacy Policy

- [X] I’m not disclosing any internal or sensitive information.

**Comment by ilhan007** [comment]
@wilson-xu-sap can you give us more information about the urgency matter described as "Expected Fixed Time: ASAP".
I hope it is not like application is breaking and not working

**Comment by wilson-xu-sap** [comment]
> @wilson-xu-sap which component, is the List?
> 
> * The List has `headerText` property, not `titleText`?
> * Do I get the request correctly that you need a new property in the List to define the heading level?

Correct. Your understanding is perfect. No heading level violates a11y standard.

**Comment by wilson-xu-sap** [comment]
> @wilson-xu-sap can you give us more information about the urgency matter described as "Expected Fixed Time: ASAP". I hope it is not like application is breaking and not working

I have received a ticket about this. This will not break the application but the ticket may be hung up for long.

**Comment by ilhan007** [comment]
Hi @wilson-xu-sap for this use-case we provide header slot, in the header slot you can pass your own custom markup as in this [example](https://sap.github.io/ui5-webcomponents/v1/play/#eyJpbmRleC5odG1sIjp7Im5hbWUiOiJpbmRleC5odG1sIiwiY29udGVudCI6IjwhLS0gcGxheWdyb3VuZC1mb2xkIC0tPlxuPCFET0NUWVBFIGh0bWw-XG48aHRtbCBsYW5nPVwiZW5cIj5cblxuXG5cbjxoZWFkPlxuICAgIFxuICAgIDxzdHlsZT5cbiAgICAgICo6bm90KDpkZWZpbmVkKSB7XG4gICAgICAgIGRpc3BsYXk6IG5vbmU7XG4gICAgICB9XG4gICAgPC9zdHlsZT5cblxuICAgIFxuICAgIDxzdHlsZT5cbiAgICAgICo6bm90KDpkZWZpbmVkKSB7XG4gICAgICAgIGRpc3BsYXk6IG5vbmU7XG4gICAgICB9XG4gICAgPC9zdHlsZT5cblxuICAgIDxtZXRhIGNoYXJzZXQ9XCJVVEYtOFwiPlxuICAgIDxtZXRhIG5hbWU9XCJ2aWV3cG9ydFwiIGNvbnRlbnQ9XCJ3aWR0aD1kZXZpY2Utd2lkdGgsIGluaXRpYWwtc2NhbGU9MS4wXCI-XG4gICAgPHRpdGxlPlNhbXBsZTwvdGl0bGU-XG48L2hlYWQ-XG5cbjxib2R5IHN0eWxlPVwiYmFja2dyb3VuZC1jb2xvcjogdmFyKC0tc2FwQmFja2dyb3VuZENvbG9yKVwiPlxuICAgIDwhLS0gcGxheWdyb3VuZC1mb2xkLWVuZCAtLT5cblxuICAgIDx1aTUtbGlzdD5cbiAgICAgIDxkaXYgc2xvdD1cImhlYWRlclwiIHN0eWxlPVwiYmFja2dyb3VuZC1jb2xvcjogdmFyKC0tc2FwR3JvdXBfVGl0bGVCYWNrZ3JvdW5kKTsgYm9yZGVyLWJvdHRvbTogMXB4IHNvbGlkIHZhcigtLXNhcEdyb3VwX1RpdGxlQm9yZGVyQ29sb3IpOyBoZWlnaHQ6IDNyZW07XG4gICAgbGluZS1oZWlnaHQ6IDNyZW07XCI-XG4gICAgICAgICA8dWk1LXRpdGxlICBsZXZlbD1cIkg0XCIgc3R5bGU9XCJmb250LWZhbWlseTogdmFyKC0tc2FwRm9udEZhbWlseSk7IHBhZGRpbmc6IDAgMXJlbTtcIj5NeSBUaXRsZTwvdWk1LXRpdGxlPlxuICAgICAgPC9kaXY-IFxuXG5cbiAgICAgICAgPHVpNS1saSBpY29uPVwibnV0cml0aW9uLWFjdGl2aXR5XCIgZGVzY3JpcHRpb249XCJUcm9waWNhbCBwbGFudCB3aXRoIGFuIGVkaWJsZSBmcnVpdFwiIGFkZGl0aW9uYWwtdGV4dD1cIkluLXN0b2NrXCJcbiAgICAgICAgICAgIGFkZGl0aW9uYWwtdGV4dC1zdGF0ZT1cIlN1Y2Nlc3NcIj5QaW5lYXBwbGU8L3VpNS1saT5cbiAgICAgICAgPHVpNS1saSBpY29uPVwibnV0cml0aW9uLWFjdGl2aXR5XCIgZGVzY3JpcHRpb249XCJPY2N1cnMgYmV0d2VlbiByZWQgYW5kIHllbGxvd1wiIGFkZGl0aW9uYWwtdGV4dD1cIkV4cGlyZXNcIlxuICAgICAgICAgICAgYWRkaXRpb25hbC10ZXh0LXN0YXRlPVwiV2FybmluZ1wiPk9yYW5nZTwvdWk1LWxpPlxuICAgICAgICA8dWk1LWxpIGljb249XCJudXRyaXRpb24tYWN0aXZpdHlcIiBkZXNjcmlwdGlvbj1cIlRoZSB5ZWxsb3cgbGVuZ3RoeSBmcnVpdFwiIGFkZGl0aW9uYWwtdGV4dD1cIlJlLXN0b2NrXCJcbiAgICAgICAgICAgIGFkZGl0aW9uYWwtdGV4dC1zdGF0ZT1cIkluZm9ybWF0aW9uXCI-Qmx1ZWJlcnJ5PC91aTUtbGk-XG4gICAgICAgIDx1aTUtbGkgaWNvbj1cIm51dHJpdGlvbi1hY3Rpdml0eVwiIGRlc2NyaXB0aW9uPVwiVGhlIHRyb3BpY2FsIHN0b25lIGZydWl0XCIgYWRkaXRpb25hbC10ZXh0PVwiUmUtc3RvY2tcIlxuICAgICAgICAgICAgYWRkaXRpb25hbC10ZXh0LXN0YXRlPVwiRXJyb3JcIj5NYW5nbzwvdWk1LWxpPlxuICAgIDwvdWk1LWxpc3Q-XG5cbiAgICA8dWk1LWxpc3QgaGVhZGVyLXRleHQ9XCJNeSBUaXRsZVwiPlxuICAgICAgICA8dWk1LWxpIGljb249XCJudXRyaXRpb24tYWN0aXZpdHlcIiBkZXNjcmlwdGlvbj1cIlRyb3BpY2FsIHBsYW50IHdpdGggYW4gZWRpYmxlIGZydWl0XCIgYWRkaXRpb25hbC10ZXh0PVwiSW4tc3RvY2tcIlxuICAgICAgICAgICAgYWRkaXRpb25hbC10ZXh0LXN0YXRlPVwiU3VjY2Vzc1wiPlBpbmVhcHBsZTwvdWk1LWxpPlxuICAgICAgICA8dWk1LWxpIGljb249XCJudXRyaXRpb24tYWN0aXZpdHlcIiBkZXNjcmlwdGlvbj1cIk9jY3VycyBiZXR3ZWVuIHJlZCBhbmQgeWVsbG93XCIgYWRkaXRpb25hbC10ZXh0PVwiRXhwaXJlc1wiXG4gICAgICAgICAgICBhZGRpdGlvbmFsLXRleHQtc3RhdGU9XCJXYXJuaW5nXCI-T3JhbmdlPC91aTUtbGk-XG4gICAgICAgIDx1aTUtbGkgaWNvbj1cIm51dHJpdGlvbi1hY3Rpdml0eVwiIGRlc2NyaXB0aW9uPVwiVGhlIHllbGxvdyBsZW5ndGh5IGZydWl0XCIgYWRkaXRpb25hbC10ZXh0PVwiUmUtc3RvY2tcIlxuICAgICAgICAgICAgYWRkaXRpb25hbC10ZXh0LXN0YXRlPVwiSW5mb3JtYXRpb25cIj5CbHVlYmVycnk8L3VpNS1saT5cbiAgICAgICAgPHVpNS1saSBpY29uPVwibnV0cml0aW9uLWFjdGl2aXR5XCIgZGVzY3JpcHRpb249XCJUaGUgdHJvcGljYWwgc3RvbmUgZnJ1aXRcIiBhZGRpdGlvbmFsLXRleHQ9XCJSZS1zdG9ja1wiXG4gICAgICAgICAgICBhZGRpdGlvbmFsLXRleHQtc3RhdGU9XCJFcnJvclwiPk1hbmdvPC91aTUtbGk-XG4gICAgPC91aTUtbGlzdD5cbiAgICA8IS0tIHBsYXlncm91bmQtZm9sZCAtLT5cbiAgICA8c2NyaXB0IHR5cGU9XCJtb2R1bGVcIiBzcmM9XCJtYWluLmpzXCI-PC9zY3JpcHQ-XG48L2JvZHk-XG5cbjwvaHRtbD5cbjwhLS0gcGxheWdyb3VuZC1mb2xkLWVuZCAtLT5cbiJ9LCJtYWluLmpzIjp7Im5hbWUiOiJtYWluLmpzIiwiY29udGVudCI6Ii8qIHBsYXlncm91bmQtaGlkZSAqL1xuaW1wb3J0IFwiLi9wbGF5Z3JvdW5kLXN1cHBvcnQuanNcIjtcbi8qIHBsYXlncm91bmQtaGlkZS1lbmQgKi9cbmltcG9ydCBcIkB1aTUvd2ViY29tcG9uZW50cy9kaXN0L0xpc3QuanNcIjtcbmltcG9ydCBcIkB1aTUvd2ViY29tcG9uZW50cy9kaXN0L1N0YW5kYXJkTGlzdEl0ZW0uanNcIjtcbmltcG9ydCBcIkB1aTUvd2ViY29tcG9uZW50cy9kaXN0L1RpdGxlLmpzXCI7XG5pbXBvcnQgXCJAdWk1L3dlYmNvbXBvbmVudHMtaWNvbnMvZGlzdC9udXRyaXRpb24tYWN0aXZpdHkuanNcIjsifX0)

The first list uses the header slot, the second list - the headerText. 
The header slot solution requires some minimal styling as well. But as you can see you can define your heading level there.
In the example I am using the ui5-title web component with some level that can be changed to your needs.
Would that  resolve the issue?

**Comment by wilson-xu-sap** [comment]
Ah, really really sorry. I have raised the PR in the wrong repo. I have re-raised an issue in ui5-webcomponent-react. Thanks for your patience and solution. Your answer will be great help for them!

**Comment by Lukas742** [comment]
Hi @wilson-xu-sap 

this was actually the correct repo as the List component is a ui5-webcomponent. We've added a note in our documentation for each component that we're just wrapping, so it's easier for developers to identify which component is a ui5-webcomponent:

>Note: This is a UI5 Web Component! [Repository](https://github.com/SAP/ui5-webcomponents) | [Documentation](https://sap.github.io/ui5-webcomponents/)

I'm going to reopen this issue now and close the one you opened in our repo.

**Comment by wilson-xu-sap** [comment]
OK. It seems that I have to create a customized header for headerText. Thanks.
